packages/memory/src/vnext.ts

```diff
@@ -0,0 +1,29 @@
+import type { MemoryConfig, SharedMemoryConfig } from '@mastra/core/memory';
+import { deepMerge } from '@mastra/core/utils';
+import { defaultWorkingMemoryTemplate, Memory as MainMemory } from './index';
+
+const newDefaults: MemoryConfig = {
+  workingMemory: {
+    enabled: false,
+    template: defaultWorkingMemoryTemplate,
+    use: 'tool-call',
+  },
+  semanticRecall: false,
+  threads: {
+    generateTitle: false,
+  },
+  lastMessages: 10,
+};
+
+export class Memory extends MainMemory {
+  constructor(config: SharedMemoryConfig = {}) {
+    super({
+      ...config,
```

Comment:
we can put a different storage/vector here if we want to. just need to move the deep merge to the whole object instead of just options